src/openpgp.js

```diff
@@ -62,28 +62,30 @@ if (globalThis.ReadableStream) {
- *                                              curve25519, p256, p384, p521, secp256k1,
- *                                              brainpoolP256r1, brainpoolP384r1, or brainpoolP512r1.
- * @param  {Date} date               (optional) override the creation date of the key and the key signatures
- * @param  {Array<Object>} subkeys   (optional) options for each subkey, default to main key options. e.g. [{sign: true, passphrase: '123'}]
- *                                              sign parameter defaults to false, and indicates whether the subkey should sign rather than encrypt
+ * Generates a new OpenPGP key pair. Supports RSA and ECC keys. By default, primary and subkeys will be of same type.
+ * @param  {String} algorithm         (optional) The primary key algorithm: ECC (default) or RSA
+ * @param  {Array<Object>} userIds    Array of user IDs e.g. [{ name:'Phil Zimmermann', email:'phil@openpgp.org' }]
+ * @param  {String} passphrase        (optional) The passphrase used to encrypt the resulting private key
+ * @param  {Number} rsaBits           (optional) Number of bits for RSA keys, defaults to 4096, or a higher value equal to config.rsaBitsMin
+ * @param  {String} curve             (optional) Elliptic curve for ECC keys:
+ *                                                curve25519 (default), p256, p384, p521, secp256k1,
+ *                                                brainpoolP256r1, brainpoolP384r1, or brainpoolP512r1
+ * @param  {Date}   date              (optional) Override the creation date of the key and the key signatures
+ * @param  {Number} keyExpirationTime (optional) Number of seconds from the key creation time after which the key expires
+ * @param  {Array<Object>} subkeys    (optional) Options for each subkey, default to main key options. e.g. [{sign: true, passphrase: '123'}]
+ *                                                sign parameter defaults to false, and indicates whether the subkey should sign rather than encrypt
  * @returns {Promise<Object>}         The generated key object in the form:
  *                                     { key:Key, privateKeyArmored:String, publicKeyArmored:String, revocationCertificate:String }
  * @async
  * @static
  */
-export function generateKey({ userIds = [], passphrase = "", rsaBits = null, keyExpirationTime = 0, curve = "curve25519", date = new Date(), subkeys = [{}] }) {
+export function generateKey({ userIds = [], passphrase = "", algorithm = "ecc", rsaBits, curve = "curve25519", keyExpirationTime = 0, date = new Date(), subkeys = [{}] }) {
   userIds = toArray(userIds);
-  curve = rsaBits ? "" : curve;
-  const options = { userIds, passphrase, rsaBits, keyExpirationTime, curve, date, subkeys };
-  if (util.getWebCryptoAll() && rsaBits && rsaBits < 2048) {
-    throw new Error('rsaBits should be 2048 or 4096, found: ' + rsaBits);
+  rsaBits = rsaBits || Math.max(4096, config.rsaBitsMin);
+  algorithm = algorithm.toLowerCase();
```

Comment:
Also I would only accept lowercase strings. Fewer variations & options that way.